src/openms_gui/source/VISUAL/TOPPViewIdentificationViewBehavior.cpp

```diff
@@ -935,24 +935,30 @@ namespace OpenMS
                               ann_spectrum[i].getIntensity());
         const String& label = labels[i];
         QColor color;
+        QColor peak_color;
+        LayerData& annotated_layer = current_canvas->getCurrentLayer();
         // XL-MS specific coloring of the labels, green for linear fragments and red for cross-linked fragments
         if (label.hasSubstring("[alpha|") || label.hasSubstring("[beta|"))
         {
           if (label.hasSubstring("|ci$"))
           {
             color = Qt::darkGreen;
+            peak_color = Qt::green;
           }
           else if (label.hasSubstring("|xi$"))
           {
             color = Qt::darkRed;
+            peak_color = Qt::red;
           }
         }
         else // different colors for left/right fragments (e.g. b/y ions)
         {
           color = (label.at(0) < 'n') ? Qt::darkRed : Qt::darkGreen;
+          peak_color = (label.at(0) < 'n') ? Qt::red : Qt::green;
         }
 
         Annotation1DItem* item = new Annotation1DPeakItem(position, label.toQString(), color);
+        annotated_layer.peak_colors_1d.push_back(peak_color);
```

Comment:
where are these deleted?

Reply:
This vector gets filled in the new LayerData object containing only the labeled peaks and labels,
this is then drawn on top of the normal spectrum.
We should delete this whole layer after changing the spectrum / ID.
`deactivate1DSpectrum` removes the annotations from the current layer, but I think a new layer with a new spectrum (and now also a filled color vector) gets created every time a new ID is selected and these do not get deleted.
@hendrikweisser can you confirm that, or is there a mechanism in place to get rid of these additional layers and spectra? Then the color vectors should be erased with them.
I will look into this later.

Reply:
@enetz: I think you're right, `deactivate1DSpectrum` and `removeTheoreticalSpectrumLayer_` (which is called by `deactivate...`) should perform the clean-up. But I'm not sure if they do everything correctly or need updating.